### PR TITLE
[ENHANCEMENT] rename-record-from-dict

### PR DESCRIPTION
**Pull request by burtenshaw**

### CODE REVIEW

File: src/argilla_sdk/records/_io/_json.py [hunk 1]
```diff
@@ -40,7 +36,7 @@ def to_json(records: List["Record"], path: Union[Path, str]) -> Path:
             path = Path(path)
         if path.exists():
             raise FileExistsError(f"File {path} already exists.")
-        record_dicts = [GenericIO._record_to_dict(record) for record in records]
+        record_dicts = [Record.to_dict(record) for record in records]
```
Comment (frascuchon) [inline src/argilla_sdk/records/_io/_json.py:39]: Does this make sense?
```suggestion
        record_dicts = GenericIO.to_list(records)
```

**Review by frascuchon — COMMENTED** [review]
Yes. This is much better as this.

The only thing (and this is something that we can review later) that is a bit "weird" to me is the usage of `_record_to_dict` method. 

We have 
- `Record.from_dict`: which does a direct map from a dictionary into a `Record` instance
- `Record.to_dict` which does the reverse action.
- `Record.from_mapping` which reads a, let's say, custom dictionary and maps into a `Record` instance by using a defined mapping (and using the `dataset.schema` as the default mapping - we should inform about this)

And it looks like the `_record_to_dict` is a sort of reverse action of the last one when `flatten=True`. I wouldn't say we should have a `Record.to_mapping`, but both methods are quite related because we can do:

```python
dict = GenericIO._record_to_dict(record)

record2 = Record.from_mapping(dict)
```

And this should work, right?

So, what if we keep `Record.from_mapping` as private for now (to don't expose the method in API refs...(`Record._from_mapping`) until we iterate a bit over mapping and flatten representations?

**Comment by burtenshaw** [comment]
> Yes. This is much better as this.
> 
> The only thing (and this is something that we can review later) that is a bit "weird" to me is the usage of `_record_to_dict` method.

My intention was for `_record_to_dict` to be different because it only used by io methods, so I didn't want it in the `Record`. 

> So, what if we keep `Record.from_mapping` as private for now (to don't expose the method in API refs...(`Record._from_mapping`) until we iterate a bit over mapping and flatten representations?

This makes sense.

**Comment by burtenshaw** [comment]
@frascuchon I moved `_record_to_dict` into `DatasetRecords` because it's the only place it is used.

**Review by frascuchon — APPROVED** [review]
Nice !

**Pull request merged**